Задание для кандидатов. Выполнить ревью.sql

```diff
@@ -0,0 +1,123 @@
+	where f.ID = @ID_Record
+		and f.FlagLoaded = cast(1 as bit)
+	)
+		begin
+			set @ErrorMessage = 'Ошибка при загрузке файла, проверьте корректность данных'
+
+			raiserror(@ErrorMessage, 3, 1)
+			return
+		end
+
+	--Чтение из слоя временных данных
+	select
+		c.ID as ID_dbo_Customer
+		,cst.ID as ID_CustomerSystemType
+		,s.ID as ID_Season
+		,cast(cs.DateBegin as date) as DateBegin
+		,cast(cs.DateEnd as date) as DateEnd
+		,c_dist.ID as ID_dbo_CustomerDistributor
+		,cast(isnull(cs.FlagActive, 0) as bit) as FlagActive
+	into #CustomerSeasonal
+	from syn.SA_CustomerSeasonal cs
+		join dbo.Customer as c on c.UID_DS = cs.UID_DS_Customer
+			and c.ID_mapping_DataSource = 1
+		join dbo.Season as s on s.Name = cs.Season
+		join dbo.Customer as c_dist on c_dist.UID_DS = cs.UID_DS_CustomerDistributor
+			and cd.ID_mapping_DataSource = 1
+		join syn.CustomerSystemType as cst on cs.CustomerSystemType = cst.Name
+	where try_cast(cs.DateBegin as date) is not null
+		and try_cast(cs.DateEnd as date) is not null
+		and try_cast(isnull(cs.FlagActive, 0) as bit) is not null
+
+	-- Определяем некорректные записи
+	-- Добавляем причину, по которой запись считается некорректной
+	select
+		cs.*
+		,case
+			when c.ID is null then 'UID клиента отсутствует в справочнике "Клиент"'
+			when c_dist.ID is null then 'UID дистрибьютора отсутствует в справочнике "Клиент"'
+			when s.ID is null then 'Сезон отсутствует в справочнике "Сезон"'
+			when cst.ID is null then 'Тип клиента отсутствует в справочнике "Тип клиента"'
+			when try_cast(cs.DateBegin as date) is null then 'Невозможно определить Дату начала'
+			when try_cast(cs.DateEnd as date) is null then 'Невозможно определить Дату окончания'
+			when try_cast(isnull(cs.FlagActive, 0) as bit) is null then 'Невозможно определить Активность'
+		end as Reason
+	into #BadInsertedRows
+	from syn.SA_CustomerSeasonal as cs
+	left join dbo.Customer as c on c.UID_DS = cs.UID_DS_Customer
+		and c.ID_mapping_DataSource = 1
+	left join dbo.Customer as c_dist on c_dist.UID_DS = cs.UID_DS_CustomerDistributor and c_dist.ID_mapping_DataSource = 1
+	left join dbo.Season as s on s.Name = cs.Season
+	left join syn.CustomerSystemType as cst on cst.Name = cs.CustomerSystemType
+	where cc.ID is null
+		or cd.ID is null
+		or s.ID is null
+		or cst.ID is null
+		or try_cast(cs.DateBegin as date) is null
+		or try_cast(cs.DateEnd as date) is null
+		or try_cast(isnull(cs.FlagActive, 0) as bit) is null
+
+	-- Обработка данных из файла
+	merge into syn.CustomerSeasonal as cs
+	using (
+		select
+			cs_temp.ID_dbo_Customer
+			,cs_temp.ID_CustomerSystemType
+			,cs_temp.ID_Season
+			,cs_temp.DateBegin
+			,cs_temp.DateEnd
+			,cs_temp.ID_dbo_CustomerDistributor
+			,cs_temp.FlagActive
+		from #CustomerSeasonal as cs_temp
+	) as s on s.ID_dbo_Customer = cs.ID_dbo_Customer
+		and s.ID_Season = cs.ID_Season
+		and s.DateBegin = cs.DateBegin
+	when matched 
+		and t.ID_CustomerSystemType <> s.ID_CustomerSystemType then
+		update

```

Comment:
Комментарий 10
неверная последовательность строк
```suggestion
                        and t.ID_CustomerSystemType <> s.ID_CustomerSystemType
	when matched then
	                update

```